sky/utils/common_utils.py

```diff
@@ -439,7 +447,8 @@ def user_and_hostname_hash() -> str:
     thus changing the SG name makes these clusters unrecognizable.
     """
     hostname_hash = hashlib.md5(socket.gethostname().encode()).hexdigest()[-4:]
-    return f'{getpass.getuser()}-{hostname_hash}'
+    username = os.getenv(constants.USER_ENV_VAR, getpass.getuser())
+    return f'{username}-{hostname_hash}'
```

Comment:
I don't think we should change this one - see the backwards incompatibility note above.